vars/commonPipelineEnvironment.groovy

```diff
@@ -91,6 +91,29 @@ class commonPipelineEnvironment implements Serializable {
         InfluxData.reset()
     }
 
+    /**
+     * Infer build tool (maven, npm, mta) based on existing build descriptor files in the project root.
+     * @param script, config
+     */
+    void inferBuildTool(script, config) {
```

Comment:
I think this function should be moved to the `setupCommonPipelineEnvironment` step.